keyboards/phantagom/oled_mpad/keymaps/default/keymap.c

```diff
@@ -0,0 +1,181 @@
+    MO(_MS), KC_MUTE, KC_VOLD, KC_VOLU,
+    KC_PSCR, KC_NUM, KC_PSLS, KC_PAST, KC_PMNS, 
+    TO(_NP), KC_HOME,   KC_UP,   KC_PGUP,   
+    TO(_MS), KC_LEFT,   _______,   KC_RIGHT,   KC_PPLS, 
+    TO(_NV), KC_END,   KC_DOWN,   KC_PGDOWN,   
+    KC_A, KC_INS, KC_DEL,  KC_PENT  
+  ),
+};
+
+void matrix_init_kb(void) {
+  // set CapsLock LED to output and low
+  setPinOutput(GP4);
+  writePinLow(GP4);
+  // set NumLock LED to output and low
+  setPinOutput(GP5);
+  writePinLow(GP5);
+  // set ScrollLock LED to output and low
+  setPinOutput(GP6);
+  writePinLow(GP6);
+}
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        writePin(GP4, led_state.num_lock);
+        writePin(GP5, led_state.caps_lock);
+        //writePin(B2, led_state.scroll_lock);
+    }
+    return res;
+}
+
+static  unsigned  int type_count = 0 ;
+ void  count_type ( void ) {
+    type_count++;
+}
+
+// Function to be called when the key is pressed/released 
+// If the function itself is not added, add each function, if it is, add only in `#ifdef OLED_ENABLE` 
+bool  process_record_user ( uint16_t keycode, keyrecord_t *record) {
+ #ifdef OLED_ENABLE 
+    if (record->event.pressed) {
+         count_type ();
+    }
+#endif 
+    return  true ;
+}
+
+void  oled_write_type_count ( void ) {
+     static  char type_count_str[ 7 ];
+     oled_write_P ( PSTR ( "Type count: " ), false );
+     itoa (type_count, type_count_str, 10 );
+     oled_write_ln (type_count_str, false );
+}
+
+// Layer state
+void  oled_write_layer_state ( void ) {
+   // Host Keyboard Layer Status
+    oled_write_P(PSTR("Layer: "), false);
+
+    switch (get_highest_layer(layer_state)) {
+        case _NUMPAD:
+            oled_write_P(PSTR("Numpad\n"), false);
+            break;
+        case _MOUSE:
+            oled_write_P(PSTR("Mouse\n"), false);
+            break;
+        case _NAV:
+            oled_write_P(PSTR("Navigation\n"), false);
+            break;
+        default:
+            // Or use the write_ln shortcut over adding '\n' to the end of your string
+            oled_write_ln_P(PSTR("Undefined\n"), false);
+    }
+}
+
+// State of NumLock/CapsLock/ScrollLock
+void  oled_write_host_led_state ( void ) {
+     const led_t led_state = host_keyboard_led_state ();
+     oled_write_P ( PSTR ( "NL:" ), false );
+     oled_write_P (led_state.num_lock ? PSTR ( "on" ) : PSTR ( "- " ), false );
+     oled_write_P ( PSTR ( " CL:" ), false );
+     oled_write_P (led_state.caps_lock ? PSTR ( "on" ) :PSTR ( "- " ), false );
+     oled_write_P ( PSTR ( " SL:" ), false );
+     oled_write_ln_P (led_state.scroll_lock ? PSTR ( "on\n" ) : PSTR ( "-\n" ), false );
+}
+
+// Uptime
+// write given digit, with leading zero if digit < 10
+void oled_write_2digit(unsigned int digit) {
+    static char buf[6];
+    itoa(digit, buf, 10);
+    if (digit < 10) {
+        oled_write_char('0', false);
+    }
+    oled_write(buf, false);
+}
+
```

Comment:
`get_u8_str` and `get_u16_str` will do this, and without using itoa, which can be expensive, size wise. 

if neither of these work, then you could simply use `get_numeric_str` instead. 

All of these are in quantum.c, and available anywhere.

Reply:
I know what you mean, but not how to solve it, not a C programmer :( but learning a lot ;)

Reply:
If you only wanted 2 digits: 
```c
const char *oled_write_2digit(uint8_t curr_num, char curr_pad) {
    static char    buf[3]   = {0};
    static uint8_t last_num = 0xFF;
    static char    last_pad = '\0';
    if (last_num == curr_num && last_pad == curr_pad) {
        return buf;
    }
    last_num = curr_num;
    last_pad = curr_pad;
    return get_numeric_str(buf, sizeof(buf), curr_num, curr_pad);
}
```

and `oled_write_2digits(uptime, '0')` would print out what you want.

Reply:
This way I am getting extra leading 0, that I don't want, it's a uptime clock.

Reply:
`oled_write_2digits(uptime, ' ')` would fix that.